### PR TITLE
✨ Authentication with Gitlab

### PR DESCRIPTION
**Pull request by wizzytod**
Users can sign in/sign up with Gitlab OAuth2 provider.
Login with gitlab

### CODE REVIEW

File: backend/src/app/http/auth/gitlab.clj [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+      (ex/raise :type :internal
+                :code :invalid-response-from-gitlab
+                :context {:status (:status res)
+                          :body (:body res)}))
+
+    (try
+      (let [data (json/read-str (:body res))]
+        (get data "access_token"))
+      (catch Throwable e
+        (log/error "unexpected error on parsing response body from gitlab access tooken request" e)
```
Comment (madmath03) [inline backend/src/app/http/auth/gitlab.clj]: Small typo, present in Google auth as well.
```suggestion
        (log/error "unexpected error on parsing response body from gitlab access token request" e)
```

File: backend/src/app/http/auth/gitlab.clj [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+                :context {:status (:status res)
+                          :body (:body res)}))
+
+    (try
+      (let [data (json/read-str (:body res))]
+        ;; (clojure.pprint/pprint data)
+        {:email (get data "email")
+         :fullname (get data "name")})
+      (catch Throwable e
+        (log/error "unexpected error on parsing response body from gitlab access tooken request" e)
```
Comment (madmath03) [inline backend/src/app/http/auth/gitlab.clj]: Small typo, present in Google auth as well.
```suggestion
        (log/error "unexpected error on parsing response body from gitlab access token request" e)
```

**Review by madmath03 — CHANGES_REQUESTED** [review]

**Review by madmath03 — APPROVED** [review]
Great work @wizzytod , this looks awesome!

**Review by niwinz — APPROVED** [review]
:heart:

**Pull request merged**